### PR TITLE
19 arm subsystem

### PR DESCRIPTION
**Pull request by LimeLlama21**
# Purpose: To have an Arm subsystem and command. The subsystem defines the functions and constants of the arm, and the command allows us to call it using the controllers.

# Files changed:
- Added the Arm subsystem and adjustArm command
- build.gradle: updated wpi.lib gradle from 2022 to 2023
- Constants.java:
   - Added ArmConstants
   - Added location of pivot point
   - Added encoder units and pivot gearbox and pulley ratios
- Moved constants from constants to:
   - Robot.java
   - Spike293Utils.java
   - SmoothControl.java
   - ArcadeDrive.java
   - ZeroArm.java
   - Arm.java

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -93,6 +90,32 @@ public static final class DrivetrainConstants {
         public static final boolean USE_NAVX_HEADING = false;
     }
 
+    public static final class ArmConstants {
```
Comment (bjmcternan) [inline src/main/java/frc/robot/Constants.java:93]: Hey delete this or move it to Arm.java

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        x += ARM_X_DELTA_MODIFIER * xPercentage;
+        y += ARM_Y_DELTA_MODIFIER * yPercentage;
+    }
+
+    public double encoderToDegrees(double encoderUnits) {
+        return encoderUnits / ENCODER_UNITS_PER_REVOLUTION * DEGREES_PER_REVOLUTION;
+    }
+
+    public void zeroEncoders() {
+        shoulderTalonFX.setSelectedSensorPosition(0);
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Arm.java:153]: Let's split this into shoulder zero and reach zero

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        reachTalonFX.setSelectedSensorPosition(0);
+    }
+
+    public void startCalibration() {
+        shoulderTalonFX.set(-CALIBRATION_MOTOR_SPEED);
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Arm.java:158]: Split into shoulder cal and reach cal

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        reachTalonFX.set(-CALIBRATION_MOTOR_SPEED);
+    }
+
+    public boolean checkCalibration() {
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Arm.java:162]: Let's think about how we will cal this. That can be another ticket

**Review by bjmcternan — CHANGES_REQUESTED** [review]
Some minor changes and we'll need to think about cal

**Comment by bjmcternan** [comment]
Don't forget to address the comments I made too

File: src/main/java/frc/robot/Robot.java [hunk 5]
```diff
@@ -12,8 +12,6 @@
 
 package frc.robot;
 
-import static frc.robot.Constants.LoggerConstants.*;
```
Comment (TreeBronch) [inline src/main/java/frc/robot/Robot.java:15]: No removals that used this. Was this just an unused import?

**Review by TreeBronch — DISMISSED** [review]
Seems fine in its context. Some imports that are removed in certain files could still be used so that should be checked but otherwise fine. Note I'm just checking for errors, not for efficiency or how good it is in a variety of contexts.

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (LimeLlama21) [reply src/main/java/frc/robot/Constants.java:93]: It was moved to Arm.java

File: src/main/java/frc/robot/subsystems/Arm.java (hunk 2 above)
Comment (LimeLlama21) [reply src/main/java/frc/robot/subsystems/Arm.java:159]: It has been split

File: src/main/java/frc/robot/Robot.java (hunk 5 above)
Comment (LimeLlama21) [reply src/main/java/frc/robot/Robot.java:15]: was removed

**Review by bjmcternan — APPROVED** [review]
This looks ok. We'll need to test this when the arm mechanism is ready and revisit this

**Review by PranjalMhetre — APPROVED** [review]
Looks Good.

**Review by bjmcternan — APPROVED** [review]

**Review by LavaWaffle — APPROVED** [review]

**Pull request merged**